### PR TITLE
Validate yaml as part of the tests.

### PR DESCRIPTION
**Pull request by samccone**

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -186,20 +191,6 @@ module.exports = function(grunt) {
       }
     },
 
-    jsDocFiles: {
```
Comment (samccone) [inline Gruntfile.js]: this is not needed since it is going to be part of the www build process :)

File: Gruntfile.js [hunk 2]
```diff
@@ -290,7 +281,7 @@ module.exports = function(grunt) {
 
   grunt.registerTask('dev', 'Auto-lints while writing code.', ['test', 'watch:marionette']);
 
-  grunt.registerTask('api', 'Build jsdoc api files', ['jsDocFiles']);
+  grunt.registerTask('api', 'test all yaml files', ['yaml-validate']);
```
Comment (samccone) [inline Gruntfile.js]: :D

File: api/collection-view.yaml [hunk 3]
```diff
@@ -334,7 +334,7 @@ functions:
       @param {Number} index
       @param {Backbone.Collection} collection
 
-   setFilter:
+    setFilter:
```
Comment (samccone) [inline api/collection-view.yaml]: @cmaher -___________-

Comment (cmaher) [inline api/collection-view.yaml]: :( this should actually just be removed.

Comment (cmaher) [inline api/collection-view.yaml]: And these spaces are really hard to get right. I'm hoping this task will tell us if they're wrong?

Comment (samccone) [inline api/collection-view.yaml]: Yep!  Hehe I'm just giving you A hard time :)

**Comment by samccone** [comment]
@cmaher we can fixup the incorrect docs before release

**Comment by jfairbank** [comment]
:+1:

File: api/collection-view.yaml (hunk 3 above)
Comment (jasonLaster) [inline api/collection-view.yaml]: How do Python devs do it

**Comment by samccone** [comment]
thanks @jasonLaster

**Pull request merged**